wgcfg/key.go

```diff
@@ -147,9 +116,9 @@ func (k *PrivateKey) String() string           { return base64.StdEncoding.Encod
 func (k *PrivateKey) HexString() string        { return hex.EncodeToString(k[:]) }
 func (k *PrivateKey) Equal(k2 PrivateKey) bool { return subtle.ConstantTimeCompare(k[:], k2[:]) == 1 }
 
-func (k *PrivateKey) IsZero() bool {
-	pk := Key(*k)
-	return pk.IsZero()
+func (k PrivateKey) IsZero() bool {
+	var zeros PrivateKey
+	return subtle.ConstantTimeCompare(zeros[:], k[:]) == 1
 }
 
 func (k *PrivateKey) clamp() {
```

Comment:
Can you add a docstring to clamp explaining what it does? The source linked in NewPrivateKey's docstring doesn't say. Going back to the original paper (https://cr.yp.to/ecdh/curve25519-20060209.pdf), this seems to just be part of the definition of a valid Curve25519 private key - a few bits here and there must be 1 or 0.

Does this get used anywhere other than in NewPrivateKey? If not, it may be less confusing to just inline the operation in NewPrivateKey and remove clamp() entirely, to make it clear that these operations are specific to generating a private key.